src/Enchantments.h

```diff
@@ -142,6 +142,9 @@ class cEnchantments
 	/** Gets random enchantment from Vector and returns it */
 	static cEnchantments GetRandomEnchantmentFromVector(cWeightedEnchantments & a_Enchantments);
```

Comment:
Couldn't you tag a_Enchantments as const and change the implementation to use a ranged for?

Reply:
`for (const auto & Enchantment : a_Enchantments)` and suchlike